pkg/instance/service.go

```diff
@@ -97,6 +97,38 @@ func (s service) SaveInstance(instance *model.DeploymentInstance) error {
 	return s.instanceRepository.SaveInstance(instance)
 }
 
+func (s service) DeleteInstance(deploymentId, instanceId uint) error {
+	deployment, err := s.FindDeploymentById(deploymentId)
```

Comment:
It would be interesting to think about the ACL. We often repeat what we already did in the handler in the service due to us doing the ACL check in the handler. How would it look if we were to move this in the service? Or how would it look if we were to pass in the entire model in the service? Something we should think about IMHO.

Reply:
We have addressed this several times. I'm not against but I feel it's beyond the scope of this PR.

Reply:
True not in this PR but we have never really addressed it and keep on writing more code than might be needed.